# Taxonomy Reference Databases

<br>

## Curated databases for Qiime and other programs

For working with 16s, 18s, or fungal ITS, there are curated databases, already formatted for Qiime (and regular versions for other programs), as well as pre-trained classifiers for standard 16S primer regions. There are details and links to these on the [**Qiime data resources page**](https://docs.qiime2.org/2019.10/data-resources/)

<br>

## Public microbiome data

The [**Qiita website**](https://qiita.ucsd.edu/) is an online resource containing data from hundreds of microbiome studies. This includes sequence data and sample metadata and allows for comparison of your study to hundreds of others. One useful utility of this site is the qiime *Clawback* plugin to create weights of your taxonomy classifier based on your study metadata (e.g. gut microbiome). There is a [**full tutorial**](https://forum.qiime2.org/t/using-q2-clawback-to-assemble-taxonomic-weights/5859) on the Qiime forum page. This is meant to increase the accuracy of your taxonomic assignments. See the 'NB-Bespoke' results in the [**Bokulich et al 2018 paper**](https://microbiomejournal.biomedcentral.com/articles/10.1186/s40168-018-0470-z) or on the [**Taxonomy classification**](https://otagoedna.github.io/2019_11_28_edna_course/chapters/taxonomy_classification.html) page of this course.

<br>

## Other Databases

The [**NCBI**](https://www.ncbi.nlm.nih.gov/) and [**EMBL**](https://www.embl.org/) sequence databases have the largest collection of sequences. Of the two, EMBL is better curated. However, both of these databases are subject to error and searches on them can be inexact and recover sequences that are not the correct gene, misidentified or unassigned taxa.

